### PR TITLE
Fix/generate wallet account

### PR DESCRIPTION
**Pull request by qfrank**
this PR should fix relate mobile [issue](https://github.com/status-im/status-mobile/issues/21096)

### CODE REVIEW

**Review by cammellos — APPROVED** [review]

File: multiaccounts/accounts/database.go [hunk 1]
```diff
@@ -1736,21 +1736,30 @@ func (db *Database) ResolveSuggestedPathForKeypair(keyUID string) (suggestedPath
...
 		found := false
 		for _, acc := range kp.Accounts {
-			if acc.Path != suggestedPath {
+			if acc.Path != suggestedPath && acc.Path != relativePath {
 				continue
 			}
 			found = true
 			break
 		}
 		if !found {
```
Comment (saledjenic) [inline multiaccounts/accounts/database.go:1753]:
```suggestion
if !found {
			// relate custom migration https://github.com/status-im/status-go/pull/5279/files#diff-e62600839ff3a2748953a173d3627e2c48a252b0a962a25a873b778313f81494
			if kp.Type == KeypairTypeProfile {
				walletRootAddress, err := db.GetWalletRootAddress()
				if err != nil {
					return relativePath, err
				}
				if kp.DerivedFrom == walletRootAddress.Hex() { // derive from wallet root
					return relativePath, nil
				}
			}
			return suggestedPath, nil
		}
```

**Review by saledjenic — APPROVED** [review]
@qfrank wondering if there is any chance to map the old derivation path format to the new one at the moment of migration from v1 to v2?

**Comment by qfrank** [comment]
> @qfrank wondering if there is any chance to map the old derivation path format to the new one at the moment of migration from v1 to v2?

can you give an example on what's the old derivation path format? @saledjenic

**Comment by saledjenic** [comment]
> can you give an example on what's the old derivation path format? @saledjenic

You tell me, the mobile app used them, not the desktop app. :)

Based on your changes, I guess, they looked like `m/0`, `m/1`, `m/2`... and now they look like `m/44'/60'/0'/0/0`, `m/44'/60'/0'/0/1`, `m/44'/60'/0'/0/2`...  so what we need to do is just to replace `m/` with `m/44'/60'/0'/0/` right?

**Comment by pavloburykh** [comment]
> > can you give an example on what's the old derivation path format? @saledjenic
> 
> You tell me, the mobile app used them, not the desktop app. :)
> 
> Based on your changes, I guess, they looked like `m/0`, `m/1`, `m/2`... and now they look like `m/44'/60'/0'/0/0`, `m/44'/60'/0'/0/1`, `m/44'/60'/0'/0/2`... so what we need to do is just to replace `m/` with `m/44'/60'/0'/0/` right?

@qfrank what I see now, is that we use old format of derivation path (`m/1`) after upgrading from v1 to v2. 

**Actual result:**

photo_2024-08-21 14 12 14


**Expected result:**

photo_2024-08-21 14 12 16

**Comment by qfrank** [comment]
> > can you give an example on what's the old derivation path format? @saledjenic
> 
> You tell me, the mobile app used them, not the desktop app. :)
> 
> Based on your changes, I guess, they looked like `m/0`, `m/1`, `m/2`... and now they look like `m/44'/60'/0'/0/0`, `m/44'/60'/0'/0/1`, `m/44'/60'/0'/0/2`... so what we need to do is just to replace `m/` with `m/44'/60'/0'/0/` right?

We can't do it in a migration, it's a special case for old mobile v1 as it have to rely on wallet root key rather than root master key after upgrading, the path will look like `m/44'/60'/0'/0/0` `m/1` , we may update relative path(e.g. `m/1`  back to `m/44'/60'/0'/0/1` when before save into db, but I doubt it make much sense, it only impact old mobile user in this case . WDYT? @saledjenic

**Comment by qfrank** [comment]
> > > can you give an example on what's the old derivation path format? @saledjenic
> > 
> > 
> > You tell me, the mobile app used them, not the desktop app. :)
> > Based on your changes, I guess, they looked like `m/0`, `m/1`, `m/2`... and now they look like `m/44'/60'/0'/0/0`, `m/44'/60'/0'/0/1`, `m/44'/60'/0'/0/2`... so what we need to do is just to replace `m/` with `m/44'/60'/0'/0/` right?
> 
> @qfrank what I see now, is that we use old format of derivation path (`m/1`) after upgrading from v1 to v2.
> 
> **Actual result:**
> 
> photo_2024-08-21 14 12 14
> 
> **Expected result:**
> 
> photo_2024-08-21 14 12 16

backend need this old format to generate the correct address, maybe we can add some modification in frontend to make it look like `m/44'/60'/0'/0/1` for user (but it will still pass the old format in backend) ?  @cammellos @pavloburykh

**Comment by saledjenic** [comment]
> it only impact old mobile user in this case

But if only that why do we do all this? 
I would say that we should maybe add SQL migration function to convert old paths to the new format, but that also requires `DerivedFrom` field to be updated.

**Comment by qfrank** [comment]
in short, SQL migration function can't fix issue 21096 , the [issue](https://github.com/status-im/status-mobile/issues/20048) description explained some reason, and this [message](https://discord.com/channels/1210237582470807632/1217172571917647932/1275547122296553617) as well :)

**Review by Samyoul — APPROVED** [review]

**Pull request merged**